src/wallet/sqlite.cpp

```diff
@@ -111,6 +111,53 @@ SQLiteDatabase::~SQLiteDatabase()
     }
 }
 
+bool SQLiteDatabase::Verify(bilingual_str& error)
+{
+    if (!PrepareDirectory()) return false;
+
+    sqlite3* db = nullptr;
+    int ret = sqlite3_open_v2(m_file_path.c_str(), &db, SQLITE_OPEN_READONLY, nullptr);
+    if (ret == SQLITE_NOTFOUND) {
+        sqlite3_close(db);
```

Comment:
From the [docs](https://www.sqlite.org/c3ref/open.html)
> Whether or not an error occurs when it is opened, resources associated with the database connection handle should be released by passing it to sqlite3_close() when it is no longer required.

Logging is handled by [global error log](https://www.sqlite.org/errlog.html)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19077#discussion_r500021272

> Logging is handled by [global error log](https://www.sqlite.org/errlog.html)

Thanks for explaining sqlite3_close. How do you know whether or not something is logged to the error log? Does every function call that doesn't return SQLITE_OK log to the error log? If so, then I guess there's no technical reason to handle errors twice even if code looks like it is silently discarding them. I would prefer the uniform approach of handling all errors, but this would give more of excuse not to.

Reply:
> Does every function call that doesn't return SQLITE_OK log to the error log?

This is my current understanding of how it works based on my tests and the following line from documentation. 

> SQLite can be configured to invoke a callback function containing an error code and a terse error message whenever anomalies occur.

I can check particular scenarios if you'd like.

Reply:
Indeed I assumed (maybe falsely) that the next maintainer is familiar with SQLite C api. My humble personal preference is not to complicate the code by duplicating the error messages, when the library can do it for us. I just tried to provide more context for the author and the reviewers, sorry if it wasn't helpful.

Reply:
From https://sqlite.org/c3ref/open.html

> Whether or not an error occurs when it is opened, resources associated with the database connection handle should be released by passing it to sqlite3_close() when it is no longer required.

Since it is ok for a database to not exist when we do `Verify`, I think it is correct to not log anything on the `NOTFOUND` error here.